src/launcher.cxx

```diff
@@ -32,7 +32,9 @@ ptr<raft_server> raft_launcher::init(ptr<state_machine> sm,
                                      ptr<logger> lg,
                                      int port_number,
                                      const asio_service::options& asio_options,
-                                     const raft_params& params_given)
+                                     const raft_params& params_given,
+                                     const raft_server::init_options opt,
+                                     cb_func::func_type raft_callback)
```

Comment:
As I mentioned in the previous PR, I'd like to suggest you to move `cb_func::func_type raft_callback` inside `raft_server::init_options` like this:
```C++
struct init_options {
    // ...

    bool skip_initial_election_timeout_;
    
    cb_func::func_type raft_callback_;
};
```
and invoke `ctx->set_cb_func(...)` is inside the constructor of `raft_server`. Then launcher doesn't need to call `set_cb_func`, thus we can get benefit for both launcher and raft_server. What do you think?

Reply:
Sorry that I didn't notice you mentioned about moving it to raft_server constructor.  I will make the change.

Actually, for this change, I am also taken reference to your unit test case:
https://github.com/eBay/NuRaft/blob/75017885cd0ad5f10c805f035fa5b44ae991aa3a/tests/unit/raft_package_fake.hxx#L85-L90